bundles/org.openhab.binding.luxtronikheatpump/src/main/java/org/openhab/binding/luxtronikheatpump/internal/LuxtronikHeatpumpHandler.java

```diff
@@ -0,0 +1,344 @@
+            case CHANNEL_EINST_BWTDI_AKT_AL:
+                value = ((DecimalType) command).intValue();
+                break;
+            case CHANNEL_BA_HZ_AKT:
+            case CHANNEL_BA_BW_AKT:
+                value = ((DecimalType) command).intValue();
+                try {
+                    // validate the value is valid
+                    HeatpumpOperationMode.fromValue(value);
+                } catch (InvalidOperationModeException e) {
+                    logger.warn("Heatpump {} mode recevieved invalid value {}: {}", channel.getCommand(), value,
+                            e.getMessage());
+                    return;
+                }
+                break;
+            case CHANNEL_EINST_WK_AKT:
+            case CHANNEL_EINST_BWS_AKT:
+            case CHANNEL_EINST_KUCFTL_AKT:
+            case CHANNEL_SOLLWERT_KUCFTL_AKT:
+                float temperature = ((DecimalType) command).floatValue();
+                value = (int) (temperature * 10);
+                break;
+            case CHANNEL_EINST_BWSTYP_AKT:
+                value = ((DecimalType) command).intValue();
+                try {
+                    // validate the value is valid
+                    HeatpumpCoolingOperationMode.fromValue(value);
+                } catch (InvalidOperationModeException e) {
+                    logger.warn("Heatpump {} mode recevieved invalid value {}: {}", channel.getCommand(), value,
+                            e.getMessage());
+                    return;
+                }
+                break;
+            case CHANNEL_EINST_KUHL_ZEIT_EIN_AKT:
+            case CHANNEL_EINST_KUHL_ZEIT_AUS_AKT:
+                float hours = ((DecimalType) command).floatValue();
+                value = (int) (hours * 10);
+                break;
+
+            default:
+                logger.debug("Received unknown channel {}", channelId);
+                break;
+        }
+
+        if (param != null && value != null) {
+            if (sendParamToHeatpump(param, value)) {
+                logger.debug("Heat pump mode {} set to {}.", channel.getCommand(), value);
+            } else {
+                logger.warn("Failed setting heat pump mode {} to {}", channel.getCommand(), value);
+            }
+        } else {
+            logger.warn("No valid value given for Heatpump operation {}", channel.getCommand());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        LuxtronikHeatpumpConfiguration config = getConfigAs(LuxtronikHeatpumpConfiguration.class);
+
+        if (!config.isValid()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "At least one mandatory configuration field is empty");
+            return;
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        synchronized (scheduledFutures) {
+            logger.debug("try to connect to the heat pump every {} seconds to update the status", RETRY_INTERVAL);
```

Comment:
Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
I'm not sure what you are trying to point out. I had a look at binding like the one for Fritz!Box while developing. It also uses logger.debug to log some stuff that is happening, like it is in this case. Shall that be removed or is there only another way it should be done?

Reply:
It's uncommon in bindings to log a periodic task has been started. You could remove it.

Reply:
ok. will remove it